rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSource.java

```diff
@@ -53,4 +64,75 @@ public interface KeyValueDataSource extends DataSource {
      * This makes things go to disk. To enable caching.
      */
     void flush();
+
+    @Nonnull
+    static KeyValueDataSource makeDataSource(@Nonnull Path datasourcePath, @Nonnull DbKind kind) {
+        String name = datasourcePath.getFileName().toString();
+        String databaseDir = datasourcePath.getParent().toString();
+
+        KeyValueDataSource ds;
+        switch (kind) {
+            case LEVEL_DB:
+                ds = new LevelDbDataSource(name, databaseDir);
+                break;
+            case ROCKS_DB:
+                ds = new RocksDbDataSource(name, databaseDir);
+                break;
+            default:
+                throw new IllegalArgumentException("kind");
+        }
+
+        ds.init();
+
+        return ds;
+    }
+
+    static void mergeDataSources(@Nonnull Path destinationPath, @Nonnull List<Path> originPaths, @Nonnull DbKind kind) {
+        Map<ByteArrayWrapper, byte[]> mergedStores = new HashMap<>();
+        for (Path originPath : originPaths) {
+            KeyValueDataSource singleOriginDataSource = makeDataSource(originPath, kind);
+            for (ByteArrayWrapper byteArrayWrapper : singleOriginDataSource.keys()) {
+                mergedStores.put(byteArrayWrapper, singleOriginDataSource.get(byteArrayWrapper.getData()));
+            }
+            singleOriginDataSource.close();
+        }
+        KeyValueDataSource destinationDataSource = makeDataSource(destinationPath, kind);
+        destinationDataSource.updateBatch(mergedStores, Collections.emptySet());
+        destinationDataSource.close();
+    }
+
+    static DbKind getDbKindValueFromDbKindLogFile(String databaseDir) {
+        try {
+            File file = new File(databaseDir, DB_KIND_PROPERTIES_FILE);
+            Properties props = new Properties();
+            if (file.exists() && file.canRead()) {
+                try (FileReader reader = new FileReader(file)) {
+                    props.load(reader);
+                }
+
+                return DbKind.ofName(props.getProperty(KEYVALUE_DATASOURCE_PROP_NAME));
+            }
+
+            return DbKind.LEVEL_DB;
+        } catch (IOException e) {
+            LoggerFactory.getLogger("KeyValueDataSource").warn(String.format("Exception found while working with %s. {}", DB_KIND_PROPERTIES_FILE), e);
+            return DbKind.LEVEL_DB;
```

Comment:
Because if the file is removed manually, then we would need to catch everytime we use this method in case we want to add default db kind. Here we are assuming that if the file is not created it will use lvel db as default unless the db is different from level db

Reply:
Done